docs/development/autoscaling-specifics-for-components.md

```diff
@@ -36,57 +36,17 @@ For both of the autoscaling modes downscaling is handled more pessimistically to
 
 ## Shoot Kubernetes API Server
 
-There are three supported autoscaling modes for the Shoot Kubernetes API server.
+The Shoot Kubernetes API server is scaled simultaneously by VPA and HPA on the same metric (CPU and memory usage).
 
-- `Baseline`
+The pod-trashing cycle between VPA and HPA scaling on the same metric is avoided by configuring the HPA to scale on average usage (not on average utilization) and by picking the target average utilization values in sync with VPA's allowed maximums.
+This makes possible VPA to first scale vertically on CPU/memory usage.
+Once all Pods' average CPU/memory usage is close to exceed the VPA's allowed maximum CPU/memory (the HPA's target average utilization, 1/7 less than VPA's allowed maximums), HPA is scaling horizontally (by adding a new replica).
```

Comment:
With https://github.com/gardener/gardener/pull/10208 kube-apiserver's maxAllowed was removed. Currently we run kube-apiserver without maxAllowed values. We have to update the docs and doc strings in the code.